src/hotspot/cpu/riscv/assembler_riscv.hpp

```diff
@@ -1933,10 +1949,10 @@ enum Nf {
     uint16_t mask = (1U << nbits) - 1;
     val <<= lsb;
     mask <<= lsb;
-    uint16_t target = *(uint16_t *)a;
+    uint16_t target = ld_c_instr(a);
     target &= ~mask;
     target |= val;
-    *(uint16_t *)a = target;
+    sd_c_instr(a, target);
```

Comment:
Compressed instructions are supposed to be located at some addresses at least 2-bytes aligned. So I am thinking that there shouldn't be any unaligned access happening here.

Reply:
on one hand - you are right, on another one - this way it looks more unified with 4-byte opcodes loads/stores